### PR TITLE
Increase plaintext&hash login speeds

### PR DESCRIPTION
**Pull request by NeffIsBack**
This PR removes unnecessary recreation of connection initialization when a login has failed, we can just reuse that smb connection. As we don't need it with kerberos (and it would be a bit more complicated) it is not changed.

Also `create_conn_obj()` now checks if we already tested smbv1 and will choose the version accordingly for further connections, to eliminate the amount of connections.

This results in double the speed when testing credentials:
Before (tested with a wordlist containing ~3,6k lines):
image
After:
image

### CODE REVIEW

File: nxc/connection.py [hunk 1]
```diff
@@ -482,6 +482,7 @@ def try_credentials(self, domain, username, owned, secret, cred_type, data=None)
             value = jitter[0] if jitter[0] == jitter[1] else random.choice(range(jitter[0], jitter[1]))
             self.logger.debug(f"Throttle authentications: sleeping {value} second(s)")
             sleep(value)
+
```
Comment (mpgn) [inline nxc/connection.py:485]: to be removed

File: nxc/protocols/smb.py [hunk 2]
```diff
@@ -309,10 +309,9 @@ def print_host_info(self):
         return True
 
     def kerberos_login(self, domain, username, password="", ntlm_hash="", aesKey="", kdcHost="", useCache=False):
-        logging.getLogger("impacket").disabled = True
```
Comment (mpgn) [inline nxc/protocols/smb.py:312]: any raison ?

File: nxc/protocols/smb.py [hunk 3]
```diff
@@ -405,7 +401,6 @@ def kerberos_login(self, domain, username, password="", ntlm_hash="", aesKey="",
 
     def plaintext_login(self, domain, username, password):
         # Re-connect since we logged off
-        self.create_conn_obj()
```
Comment (mpgn) [inline nxc/protocols/smb.py:408]: we are missing the first create_conn_obj for the first user no ?

File: nxc/protocols/smb.py (hunk 2 above)
Comment (NeffIsBack) [reply nxc/protocols/smb.py:312]: At some point it got added to the logger.py file, so it's unnecessary here:
https://github.com/Pennyw0rth/NetExec/blob/45a9121bb12c705c75c0b351e5f93ee56b985c5a/nxc/logger.py#L97

File: nxc/protocols/smb.py (hunk 3 above)
Comment (NeffIsBack) [reply nxc/protocols/smb.py:408]: It should be fine. From my experience the connection was still intact, despite the login try in `enum_host_info()`:
image

File: nxc/connection.py (hunk 1 above)
Comment (NeffIsBack) [reply nxc/connection.py]: Done

File: nxc/protocols/smb.py (hunk 3 above)
Comment (NeffIsBack) [reply nxc/protocols/smb.py:408]: Actually I should test this against a server with null session enabled, just to make sure that doesn't break it

Comment (mpgn) [reply nxc/protocols/smb.py:408]: from the comment 
> # DCs seem to want us to logoff first, windows workstations sometimes reset the connection

workstation too :)

Comment (NeffIsBack) [reply nxc/protocols/smb.py:408]: Damn it, non-dcs are not happy with that:
image

Comment (NeffIsBack) [reply nxc/protocols/smb.py:408]: Added a check in enum_host, so we create a new connection object if we want to authenticate:
image

**Review by mpgn — APPROVED** [review]

**Review by mpgn — APPROVED** [review]

**Pull request merged**